src/services/refreshToken.ts

```diff
@@ -0,0 +1,68 @@
+import axios, { AxiosError } from 'axios';
+
+import { refreshToken as authenticationRefreshToken } from 'adapters/Authentication';
+import { authenticatedHeader } from 'adapters/BaseAuth';
+import { clearToken, getToken, setToken } from 'helpers/authentication';
+import { DeserializableResponse, deserialize } from 'helpers/deserializer';
+import { paths } from 'routes';
+import { Token } from 'types/token';
+
+let isAlreadyFetchingAccessToken = false;
+let subscribers: { (isFetchSuccess: boolean): void }[] = [];
+
+function onAccessTokenFetched(isFetchSuccess: boolean) {
+  subscribers = subscribers.filter((callback) => callback(isFetchSuccess));
+}
+
+function addSubscriber(callback: { (isFetchSuccess: boolean): void }) {
+  subscribers.push(callback);
+}
+
+const doRefreshToken = (token: string) => {
+  return authenticationRefreshToken(token)
+    .then((response: DeserializableResponse) => {
+      const tokenType = deserialize<Token>(response.data);
+
+      setToken(tokenType);
+      return true;
+    })
+    .catch(() => {
+      clearToken();
+
+      // Redirect to Sign In screen
+      window.location.href = paths.signIn;
```

Comment:
The react-router-dom can help you [redirect](https://reactrouter.com/en/main/fetch/redirect)
Using it is better than the legacy way with JS 😄

Reply:
@hoangmirs good to know that we have `redirect` function on the new version 😄Unfortunately, I'm using react-router-dom v6.3.0 which doesn't support this function yet. :(